src/main/java/de/coderebell/failsafeactuator/CircuitBreakerRegistry.java

```diff
@@ -25,15 +26,11 @@
      * @param name    Which is used to identify the CircuitBreaker
      */
     public void registerCircuitBreaker(final CircuitBreaker breaker, final String name) {
-        if (name == null || name.isEmpty()) {
-            throw new IllegalArgumentException("Name for circuitbreaker needs to be set");
-        }
+        Assert.hasText(name, "Name for circuitbreaker needs to be set");
+        Assert.notNull(breaker, "Circuitbreaker to add, can't be null");
 
-        if (breaker == null) {
-            throw new IllegalArgumentException("Circuitbreaker to add, can't be null");
-        }
-
-        concurrentBreakerMap.put(name, breaker);
+        CircuitBreaker replaced = concurrentBreakerMap.put(name, breaker);
```

Comment:
Sorry, but I do not really understand this question. And so also not the use-case you want to have below.